test/core/PowerBIToken.test.ts

```diff
@@ -94,6 +95,36 @@ describe('Power BI Token', () => {
             expect(decoded.rid).to.equal(reportId);
             expect(decoded.nbf).to.equal(nbf);
             expect(decoded.exp).to.equal(exp);
+            expect(decoded.username).to.equal(username);
+        });
+        
+        it('are created with multiple RLS roles', () => {
+            let nbf = powerbi.Util.getUnixTime(new Date());
+            let exp = nbf + 3600;
+            let expiration = new Date(exp * 1000);
+            let roles = ['TestRole1', 'TestRole2'];
+
+            let embedToken = powerbi.PowerBIToken.createReportEmbedToken(workspacCollection, workspaceId, reportId, username, roles, expiration);
+            let token = embedToken.generate(accessKey);
+            let decoded = jwt.decode(token, accessKey);
+
+            expect(decoded.username).to.equal(username);
+            expect(decoded.roles).to.eql(roles);
+        });
+
+        it('fail to create when RLS username is empty and roles is not', () => {
+            let nbf = powerbi.Util.getUnixTime(new Date());
+            let exp = nbf + 3600;
+            let expiration = new Date(exp * 1000);
+            let role = 'TestRole1';
+            let badUsernames = [null, undefined, ''];
```

Comment:
My understanding is that he used `!username` to validate the claims which coerces value to boolean.  null, undefined, and '' are all falsey values and he loops through them verifying each value should cause error when there is valid role. Because they're expected to throw error they're called bad usernames